streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -466,7 +467,7 @@ void commit(final boolean startNewTransaction) {
             } else {
                 consumer.commitSync(consumedOffsetsAndMetadata);
             }
-        } catch (final CommitFailedException | ProducerFencedException error) {
+        } catch (final CommitFailedException | ProducerFencedException | TimeoutException error) {
```

Comment:
Not sure if it makes sense to convert a `TimeoutException` into a `TaskMigrated` exception... Also note that we created https://issues.apache.org/jira/browse/KAFKA-7932, so it might be best to tackle all Streams related changes there? \cc @vvcephei

Reply:
@mjsax Do you mean we have TimeoutException propagate all the way up the stack and have KAFKA-7932 handle it?

Reply:
Yeah, I agree that would be better.

Reply:
@huxihx I'd suggest we remove any Streams' related changes in this PR, since we've already created a separate JIRA (above) for tracking any follow-up works on the streams side, and @vvcephei is working on that already.

Reply:
Yes, FWIW, I agree. It would be better to just retry the operation that timed out than to kick off a rebalance.

I'll make sure there's a reference to this discussion in the ticket.